doc/examples/transform/plot_tps_deformation.py

```diff
@@ -0,0 +1,148 @@
+simulation of a 1D cubic spline, which is the solution of the biharmonic equation [2]_.
+According to [3]_, the basic solution of the biharmonic function was expanded.
+Given corresponding source and target control points, TPS is used to transform a space.
+In this case our space is a 2D image.
+
+This script shows how to use the Thine Plate Spine Transformation for
+image warping.
+
+For further information on TPS Transformation, see:
+
+.. [1] Wikipedia, Thin plate spline
+       https://en.wikipedia.org/wiki/Thin_plate_spline
+
+.. [2] Weisstein, Eric W. "Biharmonic Equation." From MathWorld--A Wolfram Web Resource
+       https://mathworld.wolfram.com/BiharmonicEquation.html
+
+.. [3] Bookstein, Fred L. "Principal warps: Thin-plate splines and the
+       decomposition of deformations." IEEE Transactions on pattern analysis and
+       machine intelligence 11.6 (1989): 567–585.
+
+
+Image Deformation
+=================
+
+In this example we will see how to use the thin plates spline interpolation to deform an image.
+To deform the image, the displacement of every pixel is needed.
+In our image, we define 6 source and destination points labelled 1-6: 1-4 in the image
+corners, 5 near the smile corner and 6 in an eye. At the corners, there are no
+displacements. The smile corner moved upward and the eye position moved down.
+
+Thin Plate Splines provides a very handy interpolator for image deformation.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import skimage as ski
+
+astronaut = ski.data.astronaut()
+
+src = np.array([[50, 50], [400, 50], [50, 400], [400, 400], [240, 150], [200, 100]])
+dst = np.array([[50, 50], [400, 50], [50, 400], [400, 400], [276, 100], [230, 100]])
+
+# Fit the thin plate spline from output to input
+
+warped_img = ski.transform.tps_warp(
+    astronaut, src[:, ::-1], dst[:, ::-1], grid_scaling=1
+)
+
+
+fig, axs = plt.subplots(1, 2)
+
+labels = [
+    '1',
+    '2',
+    '3',
+    '4',
+    '5',
+    '6',
+]  # Adjust the number of labels to match the number of points
+
+axs[0].imshow(astronaut, cmap='gray')
+axs[0].scatter(src[:, 0], src[:, 1], marker='x', color='cyan')
+
+for i, label in enumerate(labels):
+    axs[0].annotate(
+        label,
+        (src[:, 0][i], src[:, 1][i]),
+        textcoords="offset points",
+        xytext=(0, 5),
+        ha='center',
+        color='red',
+    )
+
+axs[1].imshow(warped_img, cmap='gray')
+axs[1].scatter(dst[:, 0], dst[:, 1], marker='x', color='cyan')
+
+for i, label in enumerate(labels):
+    axs[1].annotate(
+        label,
+        (dst[:, 0][i], dst[:, 1][i]),
+        textcoords="offset points",
+        xytext=(0, 5),
+        ha='center',
+        color='red',
+    )
+
+plt.show()
+
+######################################################################
+#
+# Interpolation
+# =============
+# In this example thin-plate spline is applied to each source and target
+# coordinate to derive an interpolation function and coefficients for
```

Comment:
```suggestion
# In this second example, we start with a set of source and target coordinates.
# TPS is applied to each source and target coordinate to derive an interpolation
# function and coefficients.
```
The introduction of this new example should flow better... Is there one coefficient per target point? Looking at the code, the coefficients correspond to the source/target pairs, so I'm not sure I understand 'for each target point...'